test/unit/model/validation.test.js

```diff
@@ -405,4 +405,163 @@ describe(Support.getTestDialectTeaser('InstanceValidator'), function() {
 
     });
   });
+
+  describe('custom validation functions', function () {
+
+    var User = current.define('user', {
+      age: {
+        type: Sequelize.INTEGER,
+        validate: {
+          customFn: function(val, next) {
+            if (val < 0) {
+              next('age must be greater or equal zero');
+            } else {
+              next();
+            }
+          }
+        }
+      },
+      name: Sequelize.STRING
+    }, {
+      validate: {
+        customFn: function() {
+          var user = this;
+          if (user.name === 'error') {
+            throw new Error('name must not be error');
+          }
+        }
+      }
+    });
+
+    before(function () {
+      this.stub = sinon.stub(current, 'query', function () {
+        return new Promise(function (resolve) {
+          resolve(User.build({}));
+        });
+      });
+    });
```

Comment:
Not a required change, just something you can consider for the future.

Stub could be simplified to:

`this.stub = sinon.stub(current, 'query').returns(Promise.resolve(User.build()));`
or
`this.sub = sinon.stub(current, 'query').resolves(User.build())` (assuming sinon as promised).

If model does not need to be a new instance each time.